src/app/backend/client/manager.go

```diff
@@ -279,6 +283,10 @@ func (self *clientManager) extractAuthInfo(req *restful.Request) (*api.AuthInfo,
 		return self.tokenManager.Decrypt(jweToken)
 	}
 
```

Comment:
not knowing how this method is used, does returning `nil,nil` mean the service account credentials are used instead?

if so, shouldn't the `if req == nil {` case above return an error?

Reply:
the "If request is nil then authentication will be skipped." comment on `Client(req)` sounds fragile (a call that doesn't pass in the request would escalate)

Reply:
I actually investigated our code to check if it would be possible that external request comes as nil, and it is not. `restful-go` will always initialize request object for us. The first `if req == nil` is used to initialize `insecureClient` that is used internally by our backend only and `return` at the end is there, not to break dashboard in case someone chooses to actually set `argDisableSkip` to `false`.

Reply:
> I actually investigated our code to check if it would be possible that external request comes as nil, and it is not.

That may be true at this point in time, but a code path could be introduced that could call this with a nil internally as part of building a response to satisfy an external request.

> The first if req == nil is used to initialize insecureClient (by insecure I mean client that uses dashboard SA privileges) that is used internally by our backend only

I'd like to see the internal client building made more separate from the function that is used to return clients used to handle external requests, for safety